### DISCUSSION TITLE
Method effective parameters limit

### DISCUSSION THREAD

**Discussion by AgusAzar** [microsoft/react-native-windows#13295] [Q&A] [answered]
Hi i'm doing an implementation for windows of an existing library and have some problems with a method.
The method should recieve 8 params to fit the android and ios implementation but when i try to compile it i have an error of type IndexOutOfRangeException.
After look at the implementation of CodeGenerator in Microsoft.ReactNative.Managed.CodeGen, found that a method can only have 7 named parameters.

    public static readonly SyntaxToken[] ArgLocalNames = new SyntaxToken[] {
      Arg0LocalName, Arg1LocalName, Arg2LocalName, Arg3LocalName,
      Arg4LocalName, Arg5LocalName, Arg6LocalName,
    };

Is there a reason for this limitation?

**Comment by chrisglein** [comment, marked as answer]
> Is there a reason for this limitation?

Only that the way the macros work there needed to be some limit, and this was the limit that was picked. We can extend it to cover more. I'll create an issue from this discussion to track the request.

**Reply by chrisglein** [reply]
https://github.com/microsoft/react-native-windows/issues/13307
